scipy/linalg/_basic.py

```diff
@@ -1294,16 +1294,80 @@ def pinv(a, atol=None, rtol=None, return_rank=False, check_finite=True,
     LinAlgError
         If SVD computation does not converge.
 
+    See Also
+    --------
+    pinvh : Moore-Penrose pseudoinverse of a hermititan matrix.
+
+    References
+    ----------
+    .. [1] Penrose, R. (1956). On best approximate solutions of linear matrix
+           equations. Mathematical Proceedings of the Cambridge Philosophical
+           Society, 52(1), 17-19. doi:10.1017/S0305004100030929
+
     Examples
     --------
+
+    Given an ``m x n`` matrix ``A`` and an ``n x m`` matrix ``B`` the four
+    Moore-Penrose conditions are:
+
+    1. ``ABA = A`` (``B`` is a generalized inverse of ``A``),
+    2. ``BAB = B`` (``A`` is a generalized inverse of ``B``),
+    3. ``(AB)* = AB`` (``AB`` is hermitian),
+    4. ``(BA)* = BA`` (``BA`` is hermitian) [1]_.
+
+    Here, ``A*`` denotes the conjugate transpose. The Moore-Penrose
+    pseudoinverse is a unique ``B`` that satisfies all four of these
+    conditions and exists for any ``A``. Note that, unlike the standard
+    matrix inverse, ``A`` does not have to be square or have
+    independant columns/rows.
+
+    As an example, we can calculate the Moore-Penrose pseudoinverse of a
+    random non-square matrix and verify it satisfies the four conditions.
+
     >>> import numpy as np
     >>> from scipy import linalg
     >>> rng = np.random.default_rng()
-    >>> a = rng.standard_normal((9, 6))
-    >>> B = linalg.pinv(a)
-    >>> np.allclose(a, a @ B @ a)
+    >>> A = rng.standard_normal((9, 6))
+    >>> B = linalg.pinv(A)
+
+    Condition 1:
+
+    >>> np.allclose(A @ B @ A, A)
     True
-    >>> np.allclose(B, B @ a @ B)
+
+    Condition 2:
+
+    >>> np.allclose(B @ A @ B, B)
+    True
+
+    Condition 3:
+
+    >>> np.allclose((A @ B).conj().T, A @ B)
+    True
+
+    Condition 4:
+
+    >>> np.allclose((B @ A).conj().T, B @ A)
+    True
+
+    If ``A`` is invertible then the Moore-Penrose pseudoinverse is exactly the
+    inverse of ``A`` [1]_.
```

Comment:
This can be a note, we don't need to demonstrate it. Also numerically it will cause some headache if somebody comes in with an ill-conditioned matrix and claims this line is not true. Not a big problem but just nuisance.